src/test/java/com/fasterxml/jackson/core/BaseTest.java

```diff
@@ -451,11 +451,15 @@ protected String readAndWrite(JsonFactory f, JsonParser p) throws IOException
         g.disable(JsonGenerator.Feature.AUTO_CLOSE_JSON_CONTENT);
         try {
             while (p.nextToken() != null) {
+                System.err.println(p.currentToken() + "  " + p.currentName() + "  " + p.currentValue());
                 g.copyCurrentEvent(p);
             }
         } catch (IOException e) {
             g.flush();
-            fail("Unexpected problem during `readAndWrite`. Output so far: '"+sw+"'; problem: "+e);
+            throw new AssertionError(
+                    "Unexpected problem during `readAndWrite`. Output so far: '" +
+                            sw + "'; problem: " + e.getMessage(),
+                    e);
```

Comment:
This isn't required at all but it sure made all the failures easier to see. I had a lot of failures.....